lib/EscDriver/src/EscDriverEsp32c3.h

```diff
@@ -0,0 +1,421 @@
+#ifndef _ESC_DRIVER_ESP32C3_H_
+#define _ESC_DRIVER_ESP32C3_H_
+
+#if defined(ESP32_C3)
+
+#include "EscDriver.h"
+#include <Arduino.h>
+#include <driver/rmt.h>
+
+//#define DURATION  12.5 /* flash 80MHz => minimum time unit in ns */
+static const size_t DURATION_CLOCK = 25; // [ns] duobled value to increase precision
+static const size_t ITEM_COUNT = EscDriverBase::DSHOT_BIT_COUNT + 1;
+static const int32_t DURATION_MAX = 0x7fff; // max in 15 bits
+
+#define TO_INTERVAL(v) (1 * 1000 * 1000 / (v)) // [us]
+
+// faster esc response, but unsafe (no task synchronisation)
+// set to 0 in case of issues
+#define ESPFC_RMT_BYPASS_WRITE_SYNC 1
+
+#if ESPFC_RMT_BYPASS_WRITE_SYNC
+IRAM_ATTR static esp_err_t _rmt_fill_tx_items(rmt_channel_t channel, const rmt_item32_t* item, uint16_t item_num, uint16_t mem_offset)
+{
+  //RMT_CHECK(channel < RMT_CHANNEL_MAX, RMT_CHANNEL_ERROR_STR, (0));
+  //RMT_CHECK((item != NULL), RMT_ADDR_ERROR_STR, ESP_ERR_INVALID_ARG);
+  //RMT_CHECK((item_num > 0), RMT_DRIVER_LENGTH_ERROR_STR, ESP_ERR_INVALID_ARG);
+
+  /*Each block has 64 x 32 bits of data*/
+  //uint8_t mem_cnt = RMT.conf_ch[channel].conf0.mem_size;
+  //RMT_CHECK((mem_cnt * RMT_MEM_ITEM_NUM >= item_num), RMT_WR_MEM_OVF_ERROR_STR, ESP_ERR_INVALID_ARG);
+
+  //rmt_fill_memory(channel, item, item_num, mem_offset);
+  //portENTER_CRITICAL(&rmt_spinlock);
+  RMT.sys_conf.fifo_mask = RMT_DATA_MODE_MEM;
+  //portEXIT_CRITICAL(&rmt_spinlock);
+  for(size_t i = 0; i < item_num; i++) {
+    RMTMEM.chan[channel].data32[i + mem_offset].val = item[i].val;
+  }
+  return ESP_OK;
+}
+IRAM_ATTR static esp_err_t _rmt_tx_start(rmt_channel_t channel, bool tx_idx_rst)
+{
+  //RMT_CHECK(channel < RMT_CHANNEL_MAX, RMT_CHANNEL_ERROR_STR, ESP_ERR_INVALID_ARG);
+  //portENTER_CRITICAL(&rmt_spinlock);
+  if(tx_idx_rst) {
+    RMT.tx_conf[channel].mem_rd_rst = 1;
+  }
+  //RMT.tx_conf[channel].mem_owner = RMT_MEM_OWNER_TX;
```

Comment:
not sure this is needed:
C3 RMT: 4 channels, 2 TX channels, 2 RX channels

Reply:
And huston, we have a problem here. With RMT on C3 we can drive only two outputs. I see following possibilities
1. port esp8266 implementation completely where dshot is bit-bang'ed, and analog sit on single hardware timer
2. use bin-bang dshot from esp8266 and use `ledc` driver for analog, but documentation says, that on C3 ledc uses only slow clock generator (1MHz) and it can be hard to keep accurate timinig or resolution.
So kinda lot of work and testing.

Reply:
Yes, [ESP32-C3 docu](https://docs.espressif.com/projects/esp-idf/en/latest/esp32c3/api-reference/peripherals/ledc.html) says: All the timers and channels in the ESP32-C3’s LED PWM Controller only support low speed mode. Any change of PWM settings must be explicitly triggered by software.

-ESP8266 port is not an option, the registers which in ESP8266, in ESP32-C3 do not exists, it needs a complete rewrite IMO.
-cannot open source file "user_interface.h"
![image](https://github.com/rtlopez/esp-fc/assets/7183506/47215983-d68f-4bf7-9efa-95300bcc05fe)


Maybe we should just make PWM work in the first place, then we will see? (Don't really know how to do that btw :( )